optimum/exporters/onnx/base.py

```diff
@@ -120,6 +120,7 @@ class OnnxConfig(ExportConfig, ABC):
         "seq2seq-lm": OrderedDict({"logits": {0: "batch_size", 1: "decoder_sequence_length"}}),
         "sequence-classification": OrderedDict({"logits": {0: "batch_size"}}),
         "token-classification": OrderedDict({"logits": {0: "batch_size", 1: "sequence_length"}}),
+        "speech2seq-lm": OrderedDict({"logits": {0: "batch", 1: "sequence"}}),
```

Comment:
Is it the "official" name?

We could take:

- `automatic-speech-recognition` to match the pipelines
- `speech2text`

@lewtun wdty?

Reply:
I think the idea was to partially align with the underlying [autoclass](https://huggingface.co/docs/transformers/v4.24.0/en/model_doc/auto#transformers.AutoModelForSpeechSeq2Seq), but I agree `automatic-speech-recognition` would be more intuitive.

In general (not for this PR), I think we should take the opportunity to align more closely with the Hub tasks, e.g. `seq2seq-lm` could also be `text2text-generation` right?

Reply:
Alright then I guess we can keep `speech2seq-lm` for now since the other names are aligned to the AutoClass, and maybe change that (if needed) for all the tasks in another PR.